### DISCUSSION TITLE
✨ Do not allow creating modules with reserved names

### DISCUSSION THREAD

**Discussion by sagikazarmark** [dagger/dagger#12356] [Ideas]
### What are you trying to do?

```shell
dagger init --sdk go --name module --source .
```

### Why is this important to you?

Go complains about the name `Module` being redeclared.

### How are you currently working around this?

I don't use that name, but I'm guessing other names might be affected too.

**Comment by helderco** [comment]
There's other issues covering this:
- https://github.com/dagger/dagger/issues/6952
- https://github.com/dagger/dagger/issues/6071
- https://github.com/dagger/dagger/issues/6186

**Comment by sagikazarmark** [comment]
From what I can tell, those are rather about finding a solution to resolve name conflicts. This is a proposal for the CLI (a low-hanging fruit if you will) to disallow generating modules like that in the first place.

**Comment by jedevc** [comment]
Ah, this is a very weird real problem, similarly related:

```
$ dagger -m ./version query <cmd/dagger/typedefs.graphql > out.json
```

In the resulting `out.json`, we can see that `Query.version` is updated - there's already a `Query.version` method in the core API.
